### ISSUE TITLE
Example on splitElementsOn

### ISSUE THREAD

**Issue by mustasj** [riasvdv/craft-scout#102]
Could you implement the `->splitElementsOn()` in the **Example Index Configuration**
Didn't understand how to use it...

**Comment by riasvdv** [comment]
You should only use this when you need it, when your documents are getting too large for Algolia, I've added an extra line to the README.md where it states you have to return an array for the keys you define in ->splitElementsOn

https://github.com/riasvdv/craft-scout#-splitelementsonarray-keys

**Comment by mustasj** [comment]
Some of the content is too large already.
So like this?:
```php
ScoutIndex::create( 'indexName' )
    ->elementType( Entry::class )
    ->criteria( function(EntryQuery $query) {
        return $query->site( '*' );
    } )
    ->transformer( function(Entry $entry) {

        return [
            'objectID' => $objectId,
            'title'    => $entry->title,
            'content'  => $entry->content->all(), // supertable or some array with alot of content
            'thumb'    => $thumb,
            'url'      => $entry->getUrl(),
        ];
    } )
    ->splitElementsOn( ['content'] )
    ->indexSettings(
        IndexSettings::create()
            ->distinct( true )
            ->attributeForDistinct( ['content'] )
            ->minWordSizefor1Typo( 3 )
    );
```

**Comment by riasvdv** [comment]
Yes, something like that, but you can't straight up get the elements like this `$entry->content->all()` because you're getting objects back, you need to turn them into arrays yourself with an `array_map`.

**Comment by mustasj** [comment]
Yeah i know, just an example of much content 😄 
I'll see what to do, since it is only a few entries that are too large.
Is there anyway to do this dynamicly?
maybe add a new return element like 'large_content' and `->splitElementsOn( ['large_content'] )` ?

**Comment by mark-chief** [comment]
@mustasj did you manage to do this in the end? 

@riasvdv Would also like to see an example of an `array_map`, would be super grateful! 
My matrix block textField has a ton of text which is too large for a record already. Thanks and much love for this plugin!

**Comment by mustasj** [comment]
@mark-chief no i just excluded the content that was over the limit algolia

**Comment by alexmglover** [comment]
@mark-chief and @mustasj I just figured this out and thought I could possibly help. You will need to split `content` out into an array in order to use splitElementsOn. 

I used a variant of this code to split up the content into an array: https://github.com/algolia/scout-extended/blob/2cd4d70bc0addf587a680cb83b7380b8afb16ea4/src/Splitters/HtmlSplitter.php#L69-L86

Let me know if you have any further questions and I can try to help.
